services/QuillLMS/app/services/google_integration/student_email_updater.rb

```diff
@@ -0,0 +1,48 @@
+module GoogleIntegration
+  class StudentEmailUpdater
+    attr_reader :email, :google_id
+
+    def initialize(data)
+      @email = data[:email]&.downcase
+      @google_id = data[:google_id]
+    end
+
+    def run
+      return if invalid_email?
+      return if invalid_google_id?
+      return if student_not_found?
+      return if email_not_changed?
+      return if another_user_has_email?
+
+      update_email
+    end
+
+    private def another_user_has_email?
+      ::User.where(email: email).exists?
+    end
+
+    private def email_not_changed?
+      student.email == email
+    end
+
+    private def invalid_email?
+      email.nil?
```

Comment:
You could take or leave this, but you could consider removing some of these simple methods and just use `.nil?` in the main method since it might be more clear, e.g.

```ruby
return if (email.nil? || google_id.nil? || student.nil?)

vs.

return if invalid_email?
return if invalid_google_id?
return if student_not_found?